### PR TITLE
Don't drain from session processor as it is not necessary for FIFO

### LINKED ISSUE #41807: [BUG] Azure.Messaging.ServiceBus Version 7.17.x Causing High Request Rates, Performance Issues, and Escalating Billing Costs
### Library name and version

Azure.Messaging.ServicesBus 7.17.x

### Describe the bug

we have encountered critical issue with Azure.Messaging.ServiceBus version 7.17.x, specifically related to significantly elevated request rates and performance degradation when the required session feature is enabled. The number of requests sent to Azure Service Bus has experienced a significant surge, exceeding normal levels.

Notably, we have verified that this issue does not occur with version 7.16.2.

image

### Expected behavior

Enabling required sessions in Azure.Messaging.ServiceBus version 7.17.x should not lead to a disproportionate increase in request rates or negatively impact the overall performance of the application. The expected behavior aligns with the stable performance observed in version 7.16.2.

### Actual behavior

-

### Reproduction Steps

-

### Environment

_No response_

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/41807

### CODE REVIEW

**Review by m-redding — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/SessionProcessorLiveTests.cs [hunk 1]
```diff
@@ -2679,6 +2679,54 @@ async Task ProcessMessage(ProcessSessionMessageEventArgs args)
...
+                    while (!cts.IsCancellationRequested)
+                    {
+                        await sender.SendMessageAsync(ServiceBusTestUtilities.GetMessage("session"));
+                        await Task.Delay(TimeSpan.FromMilliseconds(100));
+                    }
+                }
+
+                Task ProcessMessage(ProcessSessionMessageEventArgs args)
+                {
+                    if (args.Message.SequenceNumber != lastSequenceNumber + 1)
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/SessionProcessorLiveTests.cs:2718]: nit:  Couldn't we do the entire if as an Assert clause?  Something like:

```csharp
Assert.That(args.Message.SequenceNumber, Is.EqualTo(lastSequenceNumber + 1), $"Last sequence number: {lastSequenceNumber}, current sequence number: {args.Message.SequenceNumber}");
```

**Review by jsquire — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/SessionProcessorLiveTests.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/SessionProcessorLiveTests.cs:2718]: Hah, I don't really use that assertion style anywhere in this file, but I suppose you are right.

Comment (jsquire) [reply sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/SessionProcessorLiveTests.cs:2718]: Not meant to suggest that exact style, just what I'm used to writing.  `Assert.Equals` would do the same thing, I think.

Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/SessionProcessorLiveTests.cs]: Ah, right, yeah I picked up on that after my comment, but I still copied your code verbatim.

**Pull request merged**